frontend/src/components/transactions/ActionRowBlock.tsx

```diff
@@ -69,7 +73,8 @@ export default class extends React.Component<Props, State> {
                   <Row>
                     <Col className="action-row-timer">
                       <span className="action-row-timer-status">
-                        <ExecutionStatus status={transaction.status} />
+                        {status ? <ExecutionStatus status={status} /> : ""}
+                        {isFinal ? "/Finalized" : "/Finalizing"}
```

Comment:
Having `isFinal` as undefined (given the types I assume it is a valid case) should not be displayed as "Finalizing".

I suggest we display "Fetching Status... / Checking Finality..." when we don't have the information yet.